### PR TITLE
add detected objects message definition

### PR DESCRIPTION
**Pull request by adev4a**
# PR Details
Adds message definition for sensor detected objects.
Also updates the namespace for sensor data sharing message to include sdsm.
## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: streets_utils/streets_messages/include/detected_object_msg/cartesian_point.hpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <vector>
+#include <string>
+#include <optional>
+
+
+namespace streets_utils::messages {
```
Comment (paulbourelly999) [inline streets_utils/streets_messages/include/detected_object_msg/cartesian_point.hpp:21]: I think you can combine this into a single line
```c++
namespace streets_utils::messages::detected_object_msg
```

Comment (paulbourelly999) [reply streets_utils/streets_messages/include/detected_object_msg/cartesian_point.hpp:21]: Also if you are going to do this, could you add a message namespace for the sdsm headers, I know it maybe a lot of changes by my head says it should be relatively easy to to with simple replace all calls right?

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
If you could add namespace to the sensor data sharing msg that would be awesome.

File: streets_utils/streets_messages/include/detected_object_msg/detected_object_msg.hpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <vector>
+#include <string>
+#include <optional>
+
+#include <detection_type.hpp>
```
Comment (adev4a) [inline streets_utils/streets_messages/include/detected_object_msg/detected_object_msg.hpp:20]: This header was deleted, needs to be removed from here

**Review by paulbourelly999 — APPROVED** [review]
Looks good

**Pull request merged**